baseplate/experiments/providers/simple_experiment.py

```diff
@@ -24,6 +26,23 @@
 }
 
 
+def _generate_targeting(targeting_config):
+    """Generate the targeting tree for this experiment.
+
+    If no config is provided, then assume we want to target all. If an invalid
+    config is provided, then target none.
+    """
+    if targeting_config is None:
+        targeting_tree = create_targeting_tree({'OVERRIDE': True})
+    else:
+        try:
+            targeting_tree = create_targeting_tree(targeting_config)
+        except (TypeError, ValueError):
```

Comment:
I.E., `create_targeting_tree` should catch the `TypeError` and raise a custom exception. You also _might_ want to have it raise a custom exception rather than `ValueError` but I think that's less of a concern than the hidden `TypeError`s.